sky/global_user_state.py

```diff
@@ -390,6 +393,32 @@ def set_cluster_metadata(cluster_name: str, metadata: Dict[str, Any]) -> None:
         raise ValueError(f'Cluster {cluster_name} not found.')
 
 
+def get_cluster_storage_mounts_metadata(
+        cluster_name: str) -> Optional[Dict[str, Any]]:
+    rows = _DB.cursor.execute(
+        'SELECT storage_mounts_metadata FROM clusters WHERE name=(?)',
+        (cluster_name,))
+    for (storage_mounts_metadata,) in rows:
+        if storage_mounts_metadata is None:
+            return None
+        return pickle.loads(storage_mounts_metadata)
+    return None
+
+
+def set_cluster_storage_mounts_metadata(
+        cluster_name: str, storage_mounts_metadata: Dict[str, Any]) -> None:
```

Comment:
Same as above, should we directly pass a list of`Storage` object here, and in the function we do `pickle.dumps(storage.get_metadata())`

Reply:
Also, why is the argument here not a list of metadata? What if there are multiple storages on the cluster?

Reply:
The argument is a dictionary of {destination on remote node : storage_metadata}. It's a converted version of `storage_mounts` dictionary from `_sync_file_mounts`, which is used in `_execute_storage_mounts` and `_execute_storage_csync`. So it contains metadata of multiple storages.

The reason why we want to keep the destination as a key is to use the same `storage_mounts_metadata` and the related functions(`_set_cluster_storage_mounts_metadata`, `_get_cluster_storage_mounts_metadata`) [to relaunch mounting when `sky start` is ran after `sky stop`](https://github.com/skypilot-org/skypilot/pull/2322). We would need to restore the `storage_mounts` dictionary for this purpose. And we keep this as a metadata format to store it in `state.db`.